jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java

```diff
@@ -509,15 +509,50 @@ protected void copyRequestHeaders(HttpServletRequest clientRequest, Request prox
 
     protected void addProxyHeaders(HttpServletRequest clientRequest, Request proxyRequest)
     {
-        addViaHeader(proxyRequest);
+        addViaHeader(clientRequest, proxyRequest);
         addXForwardedHeaders(clientRequest, proxyRequest);
     }
 
+    /**
+     * Adds the HTTP Via header to the proxied request.
+     *
+     * @deprecated Use {@link #addViaHeader(HttpServletRequest, Request)} instead.
+     * @param proxyRequest the request being proxied
+     */
+    @Deprecated
     protected void addViaHeader(Request proxyRequest)
     {
         proxyRequest.header(HttpHeader.VIA, "http/1.1 " + getViaHost());
     }
 
+    /**
+     * Adds the HTTP Via header to the proxied request, taking into account data present in the client request.
+     * This method considers the protocol of the client request when forming the proxied request. If it
+     * is HTTP, then the protocol name will not be included in the Via header that is sent by the proxy, and only
+     * the protocol version will be sent. If it is not, the entire protocol (name and version) will be included. 
+     * If the client request includes a Via header, the result will be appended to that to form a chain.
+     *
+     * @since 9.4.32
```

Comment:
Remove `@since`, we don't track these things.

Reply:
Thought not. Removed.